Update structure according to methodology

razvand opened the pull request:
Update structure according to methodology:

- Use `chapters/` as top-level directory.
- Under each chapter have topics as individual directories.
- Under each topic have content types: `reading`, `drills`, `media` etc. as directories.
- Populate the directory of each content type with specific files.

razvand committed:
Update structure according to methodology

- Use `chapters/` as top-level directory.
- Under each chapter have topics as individual directories.
- Under each topic have content types: `reading`, `drills`, `media` etc.
  as directories.
- Populate the directory of each content type with specific files.
- Fix typos and formatting.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
config.yaml: Update configuration for new structure

Update `config.yaml` to fit new structure, compatible with methodology.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

teodutu commented on chapters/template-chapter/template-topic/reading/README.md:
```diff
@@ -0,0 +1,24 @@
+
+For integrating figures, you will use `![Example SVG](./media/app-os-cpu-interaction.svg)`.
+
+![Example SVG](../media/app-os-cpu-interaction.svg)
```
This image causes a builder error when running `make build`:

```
Error: Image docs/Template Chapter/Template Topic/Reading/.media/app-os-cpu-interaction.svg used in docs/Template Chapter/Template Topic/Reading/README.md not found.
    at async Promise.all (index 0)
```

VladNastase replied:
It's not the image, but the example above it. It might be a bug in Docusaurus, I need to investigate.
Since the real location of the image is one directory up, I think it is safe to also use that as an example.

VladNastase replied:
I need to mention that I also tested with `../media` instead of `/../media` in the `extra` section. I presumed that was a typo or an attempt to make the build succeed.

razvand replied:
I solved this by updating the path.

teodutu reviewed (changes requested):
I found a bug in the builder. It cannot handle the link to `../media/...` that I mentioned in my inline comment. Everything else works fine and the site is built correctly when commenting the line that includes the image.

razvand committed:
Makefile: Improve Make commands

Improve `Makefile` to make it more robust:

- Do not stop at errors when doing `clean*` actions.
- Add missing dependencies.
- Use Docker `inspect` command to check the existence of images and
  containers before cleaning.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

VladNastase commented on chapters/template-chapter/template-topic/slides/intro.md:
```diff
@@ -13,9 +13,9 @@ They also contain a list as follows:
 ### Slides including images
 
-![CPU Internals](media/cpu-internals.jpg)
+![CPU Internals](../media/cpu-internals.jpg)
```
There is a rendering issue here (file not found). I think `reveal-md` doesn't like the fact this image is in a different directory.

VladNastase replied:
I don't think there's a big problem if we duplicate/move the image to be under the `slides` directory for now. I know there might be reuse between slides and content, but we can open an issue and figure this out later.

VladNastase commented on chapters/template-chapter/template-topic/slides/content.md:
There is a typo in the table at the end of this file ("disatvantages").

VladNastase commented on config.yaml:
```diff
@@ -22,23 +22,31 @@ embed_reveal:
-          path: content/chapters/template-chapter/
-          subsections:
-            - Template-chapter/: lab/
+      - Introduction/: chapters/
```
```suggestion
      - Introduction: chapters/README.md
```

gabrielmocanu reviewed (approved):
LGTM

teodutu reviewed (approved):
:+1:
